scripts/ice-road-pipeline.py

```diff
@@ -124,15 +164,170 @@
+            # So this will be just a quick check to continue to next iter if encountered.
+            if las_name == ".DS_Store" or las_name == "ice-road":
+                continue 
+
+            rfl_fp = f'{ssa_dir}/{las_name}-rfl.las'
+
+            # Getting a translated "LAS" file
+            # "LAS" in quotations bc I am hiding the rfl here in "Z"
+            # with the intention to do fast IDW in the next step.
+            if os.path.exists(rfl_fp):
+                pass
+            else:
+                subprocess.call(["Rscript", 
+                                f"{scripts_dir}/las_ssa_prep.r", 
+                                f, crs, ni_fp, nj_fp, nk_fp, rfl_fp,
+                                n_e_d_shift,
+                                road_cal_factor,
+                                imu_data])
+
+        
+        rfl_fp = f'{ssa_dir}/rfl-merged.las'
+        rfl_fp_grid = f'{ssa_dir}/rfl-merged.tif'
+
+        # Prep PDAL commands
+        #   ZACH: resolution is 1.0 in laz2dem.py??
+        json_path = join(json_dir, 'combine-rfl.json')
+        json_pipeline = {
+            "pipeline": [
+                rfl_fp,
+                {
+                    "type":"writers.gdal",
+                    "filename":rfl_fp_grid,
+                    "resolution":pix_size, 
+                    "output_type":"idw"
+                }
+            ]
+        }
+        with open(json_path,'w') as outfile:
+            json.dump(json_pipeline, outfile, indent = 2) 
+
+        # Run PDAL commands
+        inp_str = ' '.join(glob(join(ssa_dir, '*.las')))
+        if os.path.exists(rfl_fp_grid):
+            pass    
+        else:
+            cl_call(f'pdal merge {inp_str} {rfl_fp}', log)       
+            cl_call(f'pdal pipeline {json_path}', log)     
+
+        # Prepare inputs needed for SSA raster (vectorized operation)
+        # theta_grid = 180 (perfect backscatter relative to sensor)
+        ssa_fp = f'{ssa_dir}/ssa.tif'
+        rfl_grid = rio.open_rasterio(rfl_fp_grid, masked=True)
+        rfl_grid = rfl_grid.rio.reproject_match(snowoff)
+        ssa_grid = rfl_grid.copy()
+        theta = 180 # based on data it is almost always 179-180
+        cosi=1 # because the reflectance is normalized by incidence angle in las_ssa_prep.R
+
+        # Apply ART - assuming all pixels are snow, directly solve SSA (pretty speedy)
+        k_ice = 1.8983928418833426e-06
+        wl = 1064*1e-9
+        d_ice = 917
+        g = 0.85
+        b = 1.6
+        ssa_grid = (6 * ((4 * np.pi * k_ice) / wl)) / (d_ice * (9*(1-g)) / (16*b) * (-np.log(rfl_grid / ((1.247 + 1.186 * (cosi + cosi) + 5.157 * cosi * cosi + (11.1 * np.exp(-0.087 * theta) + 1.1 * np.exp(-0.014 * theta))) / 4.0 / (cosi + cosi))))**2)
+
+        # Set threshold for snow depth and canopy height
+        # .. this is a little clunky but was having issues getting it mask both in one line..
```

Comment:
weird... the da.where((ssamask != -100) & (snowdepth > 0.08)) wasn't working?

Reply:
It was not! Which is so strange, because it worked in this way for the canopy...